pimdm/Config.py

```diff
@@ -34,18 +35,17 @@ def parse_config_file(file_path):
             pim_globals.T_LIMIT = pim_config["DefaultTimers"].get("T_LIMIT", pim_globals.T_LIMIT)
 
         if "Interfaces" in pim_config:
-            interfaces = pim_config["Interfaces"]  # type: dict
+            interface_dict = pim_config["Interfaces"]
 
-            for if_name, if_value in interfaces.items():
-                if if_value.get("ipv4", False):
-                    if_ipv4 = if_value["ipv4"]
-                    if if_ipv4.get("enabled", False):
-                        Main.add_pim_interface(interface_name=if_name, state_refresh_capable=if_ipv4.get("state_refresh", False), ipv4=True, ipv6=False)
-
-                if if_value.get("ipv6", False):
-                    if_ipv6 = if_value["ipv6"]
-                    if if_ipv6.get("enabled", False):
-                        Main.add_pim_interface(interface_name=if_name, state_refresh_capable=if_ipv6.get("state_refresh", False), ipv4=False, ipv6=True)
+            for if_name, if_value in interface_dict.items():
+                for ip_family, add_pim_interface in (
+                  ("ipv4", partial(Main.add_pim_interface, interface_name=if_name, ipv4=True, ipv6=False)),
+                  ("ipv6", partial(Main.add_pim_interface, interface_name=if_name, ipv4=False, ipv6=True)),
+                ):
+                    try:
+                        add_pim_interface(state_refresh_capable=if_value.get(ip_family, {}).get("state_refresh", False))
```

Comment:
Hey,

I didnt execute this change yet but I am wondering if it will consider the user input regarding PIM interfaces... I am asking this because just by looking to the code I get the sensation that it will add both IPv4 and IPv6 PIM interfaces regardless of the input of the user. What if the user just wants IPv4 interface? Will it add the IPv6 interface as well? Does it make sense what I am saying? I will later test this.

Reply:
Damn you are right, I have been quite lazy on testing on this PR